profileservice/api/controllers/profiles.py

```diff
@@ -621,19 +588,23 @@ def get_data_list_pid(pid):
 
 
 def check_id(id_token, data_list):
-    id_type, id_string = tokenutils.get_id_info_from_token(id_token)
+    id_string = tokenutils.get_id_info_from_token(id_token)
     auth_pass = False
 
-    if id_type == 1:  # Shibboleth ID Token
-        # get id info from data_list
-        id_from_db = data_list['uin']
-        if id_from_db == id_string:
-            auth_pass = True
-    elif id_type == 2:  # Phone ID Token
-        # get phone number from data_list
-        id_from_db = data_list['phone']
-        if id_from_db == id_string:
-            auth_pass = True
+    id_from_db = data_list['uid']
```

Comment:
Here also, backwards compatibility needs to be maintained, as `uid` won't be there in existing PII database records.

Reply:
I think these following lines should resolve this issue. Let me know if you think this will be insufficient. Thanks!
```
    id_from_db = data_list['uid']
    if id_from_db == id_string:
        auth_pass = True
    else:
        if 'uin' in data_list:  # Shibboleth ID Token
            # get id info from data_list
            id_from_db = data_list['uin']
            if id_from_db == id_string:
                auth_pass = True
        elif 'phone' in data_list:  # Phone ID Token
            # get phone number from data_list
            id_from_db = data_list['phone']
            if id_from_db == id_string:
                auth_pass = True
```

Reply:
@shurwit, For existing users, `uid` will not be present in `data_list` when GET call is made. As `uid` field is added during POST, this will result in error during GET call. Please see a subset of the stack trace below:

```
  File "rokwire-building-blocks-api/profileservice/api/controllers/profiles.py", line 620, in pii_get
    auth_pass = check_id(auth_resp, data_list[0])
  File "rokwire-building-blocks-api/profileservice/api/controllers/profiles.py", line 594, in check_id
    id_from_db = data_list['uid']
KeyError: 'uid'
```

Reply:
@shurwit, just to clarify. The above exception occurs when GET /profiles/pii is called (for existing users) before POST /profiles/pii. Thanks.

Reply:
Hi @sandeep-ps, I have made the following change which should prevent this error from occurring. Please let me know if you see other issues with this part of the code.
```
- id_from_db = data_list['uid']
+ id_from_db = data_list.get('uid')
```